airflow/jobs/scheduler_job_runner.py

```diff
@@ -1847,3 +1851,23 @@ def _executor_to_tis(self, tis: list[TaskInstance]) -> dict[BaseExecutor, list[T
             _executor_to_tis[ExecutorLoader.load_executor(executor)].append(ti)
 
         return _executor_to_tis
+
+    @provide_session
+    def _check_and_activate_dagrun_scheduling(self, session: Session = NEW_SESSION):
+        tis = session.scalars(select(TI).where(TI.state == TaskInstanceState.UP_FOR_RETRY)).all()
+        self.log.debug("Checking for DagRuns ready for scheduling")
+        drs = []
```

Comment:
```suggestion
        drs = set()
```
nit: A set would be faster with `in` operator.